### PR TITLE
15.0 mig hotel

### PR DESCRIPTION
**Pull request by fetcheping**
Hotel module migration from v14 into v15

### CODE REVIEW

File: hotel/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        access_rights_uid=None,
+    ):
+        args = args or []
+        context = self._context or {}
+        checkin_date = context.get("checkin_date")
+        checkout_date = context.get("checkout_date")
+        if isinstance(checkin_date, str):
+            checkin_date = datetime.strptime(
+                context.get("checkin_date"), tools.DEFAULT_SERVER_DATETIME_FORMAT
+            )
```
Comment (vimalpatelserpentcs) [inline hotel/models/product_product.py]: Can you please use fields.Datetime Class method for datetime conversion ?
https://github.com/odoo/odoo/blob/15.0/odoo/fields.py#L2024

File: hotel/wizard/sale_make_invoice_advance.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    def create_invoices(self):
+        ctx = self.env.context.copy()
+        if self._context.get("active_model") == "hotel.folio":
+
+            HotelFolio = self.env["hotel.folio"]
+            folio = HotelFolio.browse(self._context.get("active_ids", []))
+            folio.room_line_ids.mapped("product_id").write({"isroom": True})
+            ctx.update(
+                {
+                    "active_ids": [folio.order_id.id],
```
Comment (vimalpatelserpentcs) [inline hotel/wizard/sale_make_invoice_advance.py]:
```suggestion
                    "active_ids": folio.order_id.ids,
```

File: hotel/LICENSE/LICENSE [hunk 3]
```diff
@@ -0,0 +1,16 @@
+    OpenERP, Open Source Management Solution
+    Copyright (C) 2004-2010 OpenERP SA (<http://www.openerp.com>)
+    Copyright (C) 2021-TODAY Serpent Consulting Services Pvt. Ltd. (<http://www.serpentcs.com>).
```
Comment (vimalpatelserpentcs) [inline hotel/LICENSE/LICENSE]:
```suggestion
    Copyright (C) 2022-TODAY Serpent Consulting Services Pvt. Ltd. (<http://www.serpentcs.com>).
```

File: hotel/models/hotel_folio.py [hunk 4]
```diff
@@ -0,0 +1,838 @@
...
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
+        """
+        When you change partner_id it will update the partner_invoice_id,
+        partner_shipping_id and pricelist_id of the hotel folio as well
+        ---------------------------------------------------------------
+        @param self: object pointer
+        """
+        if self.partner_id:
```
Comment (vimalpatelserpentcs) [inline hotel/models/hotel_folio.py:243]:
```suggestion

```

File: hotel/models/hotel_folio.py [hunk 5]
```diff
@@ -0,0 +1,838 @@
...
+
+    def action_cancel(self):
+        """
+        @param self: object pointer
+        """
+        for rec in self:
+            if not rec.order_id:
+                raise UserError(_("Order id is not available"))
+            for line in rec.room_line_ids:
+                if line.order_line_id:
```
Comment (vimalpatelserpentcs) [inline hotel/models/hotel_folio.py]: Instead of if condition, add filtered function on loop.

**Comment by JayVora-SerpentCS** [comment]
@fetcheping Please squash All your commits into one at last.
thanks.

**Pull request merged**